src/Datasource/FixtureInterface.php

```diff
@@ -47,7 +49,7 @@ public function drop(ConnectionInterface $connection): bool;
      * @return \Cake\Database\StatementInterface|bool on success or if there are no records to insert,
      *  or false on failure.
      */
-    public function insert(ConnectionInterface $connection);
+    public function insert(ConnectionInterface $connection): StatementInterface|bool;
```

Comment:
Hmm.. `Datasource` packages shouldn't be using classes from `Database`.

Reply:
Removed the type hint but left the docblock annotation because this interface design is broken.